samples/dashboards/demo/personal-portfolio/demo.js

```diff
@@ -0,0 +1,429 @@
+                    ...commonMSOptions,
+                    portfolios: [
+                        portfolio
+                    ]
+                }
+            }]
+        },
+        gui: {
+            layouts: [{
+                rows: [{
+                    cells: [{
+                        id: 'kpi-wrapper',
+                        layout: {
+                            rows: [{
+                                cells: [{
+                                    id: 'kpi-holding'
+                                }, {
+                                    id: 'kpi-balance'
+                                }]
+                            }]
+                        }
+                    }]
+                }, {
+                    cells: [{
+                        id: 'wallet-chart'
+                    }]
+                }, {
+                    cells: [{
+                        id: 'data-grid'
+                    }, {
+                        id: 'kpi-gauge-risk'
+                    }]
+                },  {
+                    cells: [{
+                        id: 'goal-analysis-wrapper',
+                        layout: {
+                            rows: [{
+                                cells: [{
+                                    id: 'kpi-gauge-goal'
+                                }, {
+                                    id: 'kpi-goal-target'
+                                }, {
+                                    id: 'kpi-goal-years'
+                                }, {
+                                    id: 'kpi-goal-annual'
+                                }]
+                            }]
+                        }
+                    }]
+                }]
+            }]
+        },
+        components: [{
+            type: 'KPI',
+            renderTo: 'kpi-holding',
+            value: lastTotal,
+            valueFormat: '€{value:.2f}',
+            title: 'Holding'
+        }, {
+            type: 'KPI',
+            renderTo: 'kpi-balance',
+            value: 8000,
+            title: 'Balance',
+            valueFormat: '€{value:.2f}'
+        }, {
+            type: 'Highcharts',
+            chartConstructor: 'stockChart',
+            renderTo: 'wallet-chart',
+            chartOptions: walletChartOptions
+        }, {
+            type: 'DataGrid',
+            connector: {
+                id: 'stock-grid'
+            },
+            renderTo: 'data-grid',
+            dataGridOptions: {
+                credits: {
+                    enabled: false
+                }
+            }
+        }, {
+            renderTo: 'kpi-gauge-risk',
+            type: 'KPI',
+            connector: {
+                id: 'risk-score'
+            },
+            columnName: 'PersonalPortfolio_RiskScore',
+            chartOptions: riskScoreKPIOptions
+        }, {
+            type: 'KPI',
+            renderTo: 'kpi-goal-target',
+            value: goalAnalysisConnector.metadata.financialGoal,
```

Comment:
@ddragula @karolkolodziej - shouldn't be the `goalAnalysisConnector` loaded in connectors? Then just referenced here?

Reply:
Currently it seems it is not possible to access `metadata` from the available columns as far as I tried.

Reply:
I understand that /metadata/ is something you added in these custom connectors.

Just thinking out loud - maybe it would be worth adding it as a feature to Dashboards, so that the source of data for KPI could be some selected metadata? Only then it should probably be part of the table, not the connector, so that modifiers & all the events work also on it?

A somewhat similar feature request: https://github.com/highcharts/highcharts/issues/21559

Reply:
Also instead you can just update the KPIs after loading the board like in my [accounting demo](https://www.highcharts.com/demo/dashboards/accounting).